### ISSUE TITLE
openwsman failed with SWIG 4.1.0

### ISSUE THREAD

**Issue by jplesnik** [swig/swig#2395]
I rebuilt the latest version of SWIG (6b6761aba) successfully for [Fedora and CentOS Stream](https://copr.fedorainfracloud.org/coprs/jplesnik/swig-c9s/build/4899971/)

I tried to rebuild the dependencies and the package openwsman failed with this version.
 
```
/builddir/build/BUILD/openwsman-2.7.1/build/bindings/ruby/openwsman_wrap.c: At top level:
/builddir/build/BUILD/openwsman-2.7.1/build/bindings/ruby/openwsman_wrap.c:3906:19: error: static declaration of 'SwigClassXmlDoc' follows non-static declaration
 3906 | static swig_class SwigClassXmlDoc;
      |                   ^~~~~~~~~~~~~~~
/builddir/build/BUILD/openwsman-2.7.1/build/bindings/ruby/openwsman_wrap.c:2698:16: note: previous declaration of 'SwigClassXmlDoc' with type 'swig_class'
 2698 |     KLASS_DECL(SwigClassXmlDoc,SWIGTYPE_p__WsXmlDoc);
      |                ^~~~~~~~~~~~~~~
/builddir/build/BUILD/openwsman-2.7.1/bindings/ruby/helpers.h:54:43: note: in definition of macro 'KLASS_DECL'
   54 | #define KLASS_DECL(k,t) extern swig_class k
      |                                           ^
/builddir/build/BUILD/openwsman-2.7.1/build/bindings/ruby/openwsman_wrap.c:4367:19: error: static declaration of 'SwigClassXmlNode' follows non-static declaration
 4367 | static swig_class SwigClassXmlNode;
      |                   ^~~~~~~~~~~~~~~~
```

Complete build log is [here](https://download.copr.fedorainfracloud.org/results/jplesnik/swig-rebuild/fedora-rawhide-x86_64/04900521-openwsman/builder-live.log.gz)

**Comment by wsfulton** [comment]
Build failures don't look like they are due to recent SWIG commits. @jplesnik what was the last version of SWIG you tested openwsman with (was it in that round of tests three weeks ago?). 

Glad that @kkaempf is looking, might be due to a new Ruby version (3.1.2) being used??

**Comment by ojwb** [comment]
I didn't investigate, but just from the log snippet above it looks like a header in the openwsman project has an `extern` declaration while SWIG generates a `static` definition:

```
/builddir/build/BUILD/openwsman-2.7.1/bindings/ruby/helpers.h:54:43: note: in definition of macro 'KLASS_DECL'
   54 | #define KLASS_DECL(k,t) extern swig_class k
      |                                           ^
```

**Comment by jplesnik** [comment]
@wsfulton The build of openwsman is failing since commit 4ac3c87a2.

**Comment by jplesnik** [comment]
It seems the using of SWIGVERSION is the source of the issue.
https://github.com/Openwsman/openwsman/blob/master/bindings/ruby/helpers.h#L50

**Comment by Openwsman** [comment]
Well, it actually _is_ defined (adding `#ifdef SWIG_VERSION` to `helpers.h` proves this) but apparently has the wrong value !?
```
#if SWIG_VERSION > 0x020004
...
```
computes to `false` :exploding_head:

**Comment by Openwsman** [comment]
The problem is in `Lib/typemaps/swigversion.swg` where apparently `SWIG_VERSION_` (not the trailing underline) isn't defined.
`SWIG_VERSION` evaluates to `SWIG_VERSION_` instead of a numeric value.

**Comment by Openwsman** [comment]
Dropping the final
```
#undef SWIG_VERSION_
```
from `Lib/typemaps/swigversion.swg` fixes it.

**Comment by wsfulton** [comment]
I've reopened #1050 for discussing the best solution.

**Comment by wsfulton** [comment]
Fixed now, see 6b6761abacd5db9afcdf59a7b8019c11bff4ec5b...46f7501d94f8fb3432863344aeda6d4fd85fb1a4

**Comment by kkaempf** [comment]
Confirmed fixed, thanks !
